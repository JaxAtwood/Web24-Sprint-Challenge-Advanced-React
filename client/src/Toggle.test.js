import React from "react"
import { render, cleanup, fireEvent } from "@testing-library/react";
import '@testing-library/jest-dom/extend-expect';
// import { getByDisplayValue } from "@testing-library/dom";
import Toggle from "./hooks/Toggle";

afterEach(cleanup);

it("renders", () => {
    const { asFragment } = render(<Toggle text="false" />); 
    expect(asFragment()).toMatchSnapshot();
});

it("tests", () => {
    const { getByTestId, getByText } = render(<Toggle />);

    expect(getByTestId("divvy")).toHaveClass("toggle");
    fireEvent.click(getByText("Toggle"))

});

// const { getByLabelText } = render(<Toggle />);
// const container = document.body
// const inputNode = getByLabelText(container, 'toggle', {
//   selector: 'onClick',
// })