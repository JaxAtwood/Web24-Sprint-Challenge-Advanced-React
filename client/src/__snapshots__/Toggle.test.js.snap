// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div>
    <div
      class="color-mode__toggle"
    >
      <div
        class="toggle"
        data-testid="divvy"
      />
      Toggle
    </div>
  </div>
</DocumentFragment>
`;
